Source/FluentSpreadsheets/Sheets/SegmentImplementations/HeaderRowFooterSegmentBase.cs

```diff
@@ -0,0 +1,26 @@
+using FluentSpreadsheets.Sheets.Segments;
+using FluentSpreadsheets.Sheets.Visitors;
+
+namespace FluentSpreadsheets.Sheets;
+
+public abstract class HeaderRowFooterSegmentBase<THeaderData, TRowData, TFooterData> :
+    IHeaderRowFooterSegment<THeaderData, TRowData, TFooterData>
+{
+    public abstract IComponent BuildHeader(THeaderData data);
+
+    public abstract IComponent BuildRow(TRowData data, int rowIndex);
+
+    public abstract IComponent BuildFooter(TFooterData data);
+
+    public virtual void Accept(ISheetSegmentVisitor visitor)
+    {
+        if (visitor is IHeaderSegmentVisitor<THeaderData> headerBuilderVisitor)
+            headerBuilderVisitor.Visit(this);
+        
+        if (visitor is IRowSegmentVisitor<TRowData> rowBuilderVisitor)
+            rowBuilderVisitor.Visit(this);
+        
+        if (visitor is IFooterSegmentVisitor<TFooterData> footerBuilderVisitor)
+            footerBuilderVisitor.Visit(this);
```

Comment:
Looks a bit clumsy, I agree, but the point here is to separate header, row and footer builders, so to put all these methods in one interface, it would have a 3 generic parameters. That means we would have to pass a type arguments to type of visitor we accept in segment's `Accept` method. But we have no idea what are type arguments for other segment parts, so it is not possible to union these methods, at least I haven't come around on how to do that.

Reply:
Okay, understandable